### PR TITLE
refactor to allow MSVC without #define NOMINMAX

### PR DESCRIPTION
**Pull request by Dekken**
fix for https://github.com/jbeder/yaml-cpp/issues/904

reference: https://github.com/microsoft/cppwinrt/issues/479

### CODE REVIEW

File: include/yaml-cpp/node/convert.h [hunk 1]
```diff
@@ -120,15 +120,16 @@ typename std::enable_if<(std::is_same<T, unsigned char>::value ||
...
+    if (num >= (std::numeric_limits<T>::min)() &&
+        num <= (std::numeric_limits<T>::max)()) {
       rhs = num;
       return true;
     }
   }
   return false;
 }
 
+
```
Comment (jbeder) [inline include/yaml-cpp/node/convert.h:132]: Stray extra newline?

Comment (Dekken) [reply include/yaml-cpp/node/convert.h:132]: oh yes one moment

Comment (Dekken) [reply include/yaml-cpp/node/convert.h]: fix

Comment (Dekken) [reply include/yaml-cpp/node/convert.h]: fixed

**Pull request merged**